docs/internals/optimizer.rst

```diff
@@ -1121,6 +1122,46 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and memory store statements if conditions are met for said removal.
+In case of an ``sstore``, if all outgoing code paths revert (due to an explicit ``revert()``, ``invalid()``, or infinite recursion) or
+lead to another ``sstore`` for which the optimizer can tell that it will overwrite the first store, the statement will be removed.
```

Comment:
We should probably filll in a bit more detail here. It may be obvious, but between the initial ``sstore`` and the revert or the overwriting ``sstore`` there also cannot be any statement that could possibly read the value back that was initially written.

That includes external calls, user-defined functions with *any* storage access and ``sload``s of a slot that can't be proven to differ from the slot written by the initial ``sstore``.